src/js/ui/DataStorage.js

```diff
@@ -0,0 +1,24 @@
+export default class DataStorage {
+  #key;
+
+  /**
+   *
+   * @param {string} key
+   */
+  constructor(key) {
+    this.#key = key;
+  }
+
+  /**
+   *
+   * @param {Object|Object[]} data
+   */
+  saveItem(data) {
+    localStorage.setItem(this.#key, JSON.stringify(data));
+  }
+
+  loadItem() {
+    const stringified = localStorage.getItem(this.#key);
+    return stringified ? JSON.parse(stringified) : null;
+  }
```

Comment:
```suggestion
  #key; #storage;

  /**
   *
   * @param {string} key
   */
  constructor(key, storage = localStorage) {
    this.#key = key;
    this.#storage = storage;
  }

  /**
   *
   * @param {Object|Object[]} data
   */
  saveItem(data) {
    this.#storage.setItem(this.#key, JSON.stringify(data));
  }

  loadItem() {
    const stringified = this.#storage.getItem(this.#key);
    return stringified ? JSON.parse(stringified) : null;
  }
```
이런식으로 사용한다면 storage에 localStorage가 올 수도 있고, sessionStorage가 올 수도 있고, 혹은 내가 직접 만든 storage가 올 수도 있겠죠!?